src/components/schedule-visualizer/svg-text.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+export interface SvgTextProps extends React.SVGProps<SVGTextElement> {
+  text: string;
+  targetWidth?: number;
+}
+
+export default function SvgText(props: SvgTextProps): React.ReactElement {
+  const { text, targetWidth, ...otherProps } = props;
+  const callbackRef = (textElem: SVGTextElement | null) => {
+    if (!textElem) {
+      return;
+    }
+
+    if (!targetWidth) {
+      textElem.textContent = text;
+      return;
+    }
+    // svg text does not support auto ellipses, this workaround by testing the text length and
+    // truncate it bit by bit until it fits the icon. Its abit hacky but it shouldn't be too bad
+    // unless the robot name is mega long.
+    for (textElem.textContent = text; textElem.getComputedTextLength() > targetWidth; ) {
+      textElem.textContent =
+        textElem.textContent!.slice(0, textElem.textContent!.length - 6) + '...';
+    }
+  };
```

Comment:
> The test is failing because getComputedTextLength() is not supported in enzyme/jsdom. I tried mocking React.useRef but it doesn't work because enzyme overwrites the mock value after it renders the component. What do you think is a better approach to fix this?
> * move robot and robot-overlay tests to e2e, it will run in a real dom.
> * refactor the text svg out of robot and add a optional prop to customize the text, in unit tests, we will use a dummy component.

What do you think about adding a `try` and `catch` here avoid the extra props in `robot.tsx` and `robots-overlay`?

Reply:
Thanks for the suggestion, I now fallback to the previous implementation if `getComputedTextLength` is not a function.